x-pack/plugins/transform/server/lib/alerting/transform_health_rule_type/transform_health_service.ts

```diff
@@ -97,16 +98,36 @@ export function transformHealthServiceProvider({
       resultTransformIds = resultTransformIds.filter((id) => !excludeIdsSet.has(id));
     }
 
-    return resultTransformIds;
+    return new Set(resultTransformIds);
   };
 
-  const getTransformStats = memoize(async (transformIds: string[]): Promise<TransformStats[]> => {
-    return (
-      await esClient.transform.getTransformStats({
-        transform_id: transformIds.join(','),
-      })
-    ).transforms as TransformStats[];
-  });
+  const getTransformStats = memoize(
+    async (transformIds: Set<string>): Promise<TransformStats[]> => {
+      // Result list of transform IDs can be too long, so we need to split them into chunks.
+      const maxPathComponentLength = 2000;
```

Comment:
nit: this can be a static constant outside of this function

Reply:
moved in https://github.com/elastic/kibana/pull/183153/commits/f6afc5348551971cc2a64d8eb73e914f2e040ad3